konfera/models/schedule.py

```diff
@@ -18,8 +22,41 @@ class Schedule(KonferaModel):
             MaxValueValidator(300),
             MinValueValidator(0)
         ],
-        help_text=_('Duration in minutes.'))
+        help_text=_('Duration in minutes. If talk is selected, value will be generated.'))
     room = models.ForeignKey('Room', blank=True, null=True, related_name='scheduled_rooms')
 
     def __str__(self):
         return _('%(start)s (%(duration)s min)') % {'start': self.start, 'duration': self.duration}
+
+    def clean(self, *args, **kwargs):
+        # Only approved talk can be scheduled
+        if self.talk and self.talk.status != self.talk.APPROVED:
+            raise ValidationError({'talk': _('You cannot schedule unapproved talks.')})
+
+        # Make sure date and time is within the event's range
+        if self.event.date_from > self.start or self.start > self.event.date_to:
```

Comment:
`self.start > self.event.date_to`

Shouldn't it be `self.start + datetime.timedelta(minutes=self.duration) > self.event.date_to` ? 

Otherwise it will be possible to set a schedule, for a 40 minute long talk, to 1 minute before the event ends.